GNS3/gns3-gui
# Run IOS devices on a cloud instance

This branch adds the ability to upload an IOS image to cloud files and then run it on a cloud instance.

To test:
- Go to preferences->dynamips->images, and add an image of "cloud type".  Image will commence uploading to Cloud Files.
- Once image is uploaded, in a cloud project with a running cloud instance, drop a device of the image type above onto the canvas.  The image will start running on the cloud instance.

Known issues:
- Cloud instances are stored in the .ini file.  It currently ends up with duplicates of the same instance.
- When a new cloud instance is created, the status in the cloud inspector goes green before the instance is fully ready.  Watch the log for a steady state before dropping a device on the canvas.
- Saving and opening cloud projects is not yet working.

Given these issues, you should probably merge manually so that you can re-disable the uploading of cloud images for the time being, similar to what we did with the other cloud functionality.
